### PR TITLE
fix detached head after submodule init

### PR DESCRIPTION
**Pull request by Himanshu-Singh-Chauhan**
improve for windows script - `setup_environment.ps1`

changes - 
- `git submodule update` leaves a detached HEAD, fix that. (`.sh` scripts already have it)
- change order of symbolic link creation. - [context discord message](https://discord.com/channels/1237009981803462729/1249121705419608205/1258075352274702416) by @MaySoMusician

submodule update leaves detached head in submodule directory, and then proceeds the rest of the script with that detached head, we don't install node modules from latest changes, we need to check out to main branch and pull latest changes and run the script on main branch. mac script already does this, this pr adds it for windows scripts too.

the very first run of scripts, right after cloning repo, symlink creation fails silently because submodule directory is empty and is initialized after it.

### CODE REVIEW

**Comment by cqdev-co** [comment]
Don't worry about the macOS failing, that's definitely a oversight by me when I included that in there

**Review by cqdev-co — COMMENTED** [review]
This is a little bit out of my element, but from what I understand it from it, it looks pretty good!

**Comment by Himanshu-Singh-Chauhan** [comment]
> This is a little bit out of my element, but from what I understand it from it, it looks pretty good!

np, I tested it myself, it works.

File: scripts/pearai/setup-environment.ps1 [hunk 1]
```diff
@@ -19,29 +19,44 @@ function Initialize-BaseFunctionality {
...
+    Write-Host "`nSetting the submodule directory to match origin/main's latest changes..." -ForegroundColor White
+    Invoke-CMD -Command "git reset origin/main" -ErrorMessage "Failed to git reset to origin/main"
+    Invoke-CMD -Command "git reset --hard" -ErrorMessage "Failed to reset --hard"
+    Write-Host "`nChecking out the 'main' branch in submodule..." -ForegroundColor White
+    Invoke-CMD -Command "git checkout main" -ErrorMessage "Failed to checkout the 'main' branch in the submodule"
+    Invoke-CMD -Command "git fetch origin" -ErrorMessage "Failed to fetch latest changes from origin"
+    Invoke-CMD -Command "git pull origin main" -ErrorMessage "Failed to pull latest changes from origin/main"
+
+
 	$script = Join-Path -Path $modulePath -ChildPath 'scripts\install-dependencies.ps1'
```
[the remark this answers is not in the document]
Comment (Himanshu-Singh-Chauhan) [reply scripts/pearai/setup-environment.ps1:36]: I think the indentation is correct here, the line is inside a function.
it looks confusing when you see changes or compare diff in browser.

[the remark this answers is not in the document]
Comment (MaySoMusician) [reply scripts/pearai/setup-environment.ps1:36]: image

L36 is too indented?

[the remark this answers is not in the document]
Comment (Himanshu-Singh-Chauhan) [reply scripts/pearai/setup-environment.ps1:36]: but somehow it looks correct in vscode, 😆  
this is the diff of these same changes, 
anyways, if its still indented after merging, will fix it.
image

[the remark this answers is not in the document]
Comment (MaySoMusician) [reply scripts/pearai/setup-environment.ps1:36]: i think spaces and tabs mixed up there ... ?

[the remark this answers is not in the document]
Comment (nang-dev) [reply scripts/pearai/setup-environment.ps1:36]: @Himanshu-Singh-Chauhan

**Comment by nang-dev** [comment]
retrying macOS ci

File: scripts/pearai/setup-environment.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (nang-dev) [reply scripts/pearai/setup-environment.ps1:36]: okay lets see after merge

**Comment by nang-dev** [comment]
@cqdev-co  macOS CI still having issues, would you mind taking a quick look? :D

**Pull request merged**